cri/v1alpha1/cri_utils.go

```diff
@@ -822,17 +823,19 @@ func imageToCriImage(image *apitypes.ImageInfo) (*runtime.Image, error) {
 // ensureSandboxImageExists pulls the image when it's not present.
 func (c *CriManager) ensureSandboxImageExists(ctx context.Context, imageRef string) error {
 	_, _, _, err := c.ImageMgr.CheckReference(ctx, imageRef)
-	// TODO: maybe we should distinguish NotFound error with others.
 	if err == nil {
 		return nil
 	}
 
-	err = c.ImageMgr.PullImage(ctx, imageRef, nil, bytes.NewBuffer([]byte{}))
-	if err != nil {
-		return fmt.Errorf("pull sandbox image %q failed: %v", imageRef, err)
+	if errtypes.IsNotfound(err) {
+		err = c.ImageMgr.PullImage(ctx, imageRef, nil, bytes.NewBuffer([]byte{}))
+		if err != nil {
+			return fmt.Errorf("failed to pull sandbox image %q: %v", imageRef, err)
+		}
+		return nil
 	}
 
-	return nil
+	return fmt.Errorf("failed to checkReference sandbox image %q: %v", imageRef, err)
```

Comment:
I think we should use other word instead of `checkReference`. how about `failed to check sanbox image`?